pkg/trace/api/api.go

```diff
@@ -399,13 +409,34 @@ func decodeTraces(v Version, req *http.Request) (pb.Traces, error) {
 		}
 		var traces pb.Traces
 		err := traces.UnmarshalMsgDictionary(buf.Bytes())
-		return traces, err
+		return &pb.TracerPayload{
+			LanguageName:    ts.Lang,
+			LanguageVersion: ts.LangVersion,
+			ContainerID:     req.Header.Get(headerContainerID),
+			Chunks:          traceChunksFromTraces(traces),
+			TracerVersion:   ts.TracerVersion,
+		}, err
+	case v06:
+		buf := getBuffer()
+		defer putBuffer(buf)
+		if _, err := io.Copy(buf, req.Body); err != nil {
+			return nil, err
+		}
+		var tracerPayload pb.TracerPayload
+		err := tracerPayload.Unmarshal(buf.Bytes())
```

Comment:
We've worked quite hard in https://github.com/DataDog/datadog-agent/pull/5965 to add the `v0.5` format with a string dictionary. I wouldn't want us to regress here and throw all that away. It gives us a huge performance benefit.

Can we please implement a decoder which uses a dictionary of strings exactly like in the PR? You would have (please check the description in #5965) a payload from the tracer that is an array containing 2 elements:
* An array of strings, representing every string in the `TracerPayload` structure.
* The `TracerPayload` structure itself, where each string is replaced by an integer pointing to the index in the dictionary where the corresponding string can be found.

This should not be very hard to do, as all helper functions are already available.